lib/core/src/display/shape/text/render.rs

```diff
@@ -0,0 +1,202 @@
+//! Rendering TextField.
+
+pub mod assignment;
+pub mod selection;
+
+use crate::prelude::*;
+
+use crate::display::object::DisplayObjectData;
+use crate::display::shape::glyph::font::FontRegistry;
+use crate::display::shape::glyph::font::FontRenderInfo;
+use crate::display::shape::glyph::system::GlyphSystem;
+use crate::display::shape::text::content::TextFieldContent;
+use crate::display::shape::text::content::TextFieldContentFullInfo;
+use crate::display::shape::text::cursor::Cursor;
+use crate::display::shape::text::cursor::Cursors;
+use crate::display::shape::text::render::assignment::GlyphLinesAssignment;
+use crate::display::shape::text::render::assignment::LineFragment;
+use crate::display::shape::text::render::selection::SelectionSpritesGenerator;
+use crate::display::shape::primitive::def::*;
+use crate::display::shape::primitive::system::ShapeSystem;
+use crate::display::symbol::geometry::compound::sprite::Sprite;
+
+use nalgebra::Vector2;
+use nalgebra::Vector3;
+use crate::display::shape::text::TextFieldProperties;
+use crate::display::world::World;
+
+
+// =======================
+// === RenderedContent ===
+// =======================
+
+type GlyphLine = crate::display::shape::glyph::system::Line;
+
+/// Structure containing sprites bound to one cursor with its selection.
+#[derive(Debug)]
+pub struct CursorSprites {
```

Comment:
The name CursorSprites do not mention selection. It is misleading imo. I know this is not very helpful, I cannot find better name now, but this is not the best one.

Reply:
I assumend everywhere, that every cursor has possibly selection bound to it. See `Cursor` struct for instance